Realm/Realm.SourceGenerator/Analytics.cs

```diff
@@ -0,0 +1,524 @@
+            { SdkFeature.ReferenceDictionary, 0 },
+            { SdkFeature.PrimitiveDictionary, 0 },
+            { SdkFeature.ReferenceSet, 0 },
+            { SdkFeature.PrimitiveSet, 0 },
+            { SdkFeature.RealmInteger, 0 },
+            { SdkFeature.RealmObjectReference, 0 },
+            { SdkFeature.RealmValue, 0 },
+            { SdkFeature.GetInstanceAsync, 0 },
+            { SdkFeature.GetInstance, 0 },
+            { SdkFeature.FIXME, 0 },
+            { SdkFeature.Find, 0 },
+            { SdkFeature.WriteAsync, 0 },
+            { SdkFeature.ThreadSafeReference, 0 },
+            { SdkFeature.FIXME_TWO, 0 },
+            { SdkFeature.ShouldCompactOnLaunch, 0 },
+            { SdkFeature.MigrationCallback, 0 },
+            { SdkFeature.RealmChanged, 0 },
+            { SdkFeature.ListSubscribeForNotifications, 0 },
+            { SdkFeature.SetSubscribeForNotifications, 0 },
+            { SdkFeature.DictionarySubscribeForNotifications, 0 },
+            { SdkFeature.ResultSubscribeForNotifications, 0 },
+            { SdkFeature.PropertyChanged, 0 },
+            { SdkFeature.RecoverOrDiscardUnsyncedChangesHandler, 0 },
+            { SdkFeature.RecoverUnsyncedChangesHandler, 0 },
+            { SdkFeature.DiscardUnsyncedChangesHandler, 0 },
+            { SdkFeature.ManualRecoveryHandler, 0 },
+            { SdkFeature.GetProgressObservable, 0 },
+            { SdkFeature.PartitionSyncConfiguration, 0 },
+            { SdkFeature.FlexibleSyncConfiguration, 0 },
+            { SdkFeature.Anonymous, 0 },
+            { SdkFeature.EmailPassword, 0 },
+            { SdkFeature.Facebook, 0 },
+            { SdkFeature.Google, 0 },
+            { SdkFeature.Apple, 0 },
+            { SdkFeature.JWT, 0 },
+            { SdkFeature.ApiKey, 0 },
+            { SdkFeature.ServerApiKey, 0 },
+            { SdkFeature.Function, 0 },
+            { SdkFeature.CallAsync, 0 },
+            { SdkFeature.FIX_ME_THREE, 0 },
+            { SdkFeature.DynamicApi, 0 },
+        };
+
+        private Dictionary<string, string> _realmEnvMetrics = new()
+        {
+            { SdkFeature.UserId, string.Empty },
+            { SdkFeature.RealmSdk, ".NET" },
+            { SdkFeature.Language, "C#" },
+            { SdkFeature.HostOsType, string.Empty },
+            { SdkFeature.HostOsVersion, string.Empty },
+        };
+
+        private static string AnonymizedUserID
+        {
+            get
+            {
+                try
+                {
+                    var id = GenerateComputerIdentifier();
+                    return id != null ? SHA256Hash(id) : "UNKNOWN";
+                }
+                catch
+                {
+                    return "UNKNOWN";
+                }
+            }
+        }
+
+        public void SubmitAnalytics(GeneratorExecutionContext context)
+        {
+            var payload = string.Empty;
+
+            if (Environment.GetEnvironmentVariable("REALM_DISABLE_ANALYTICS") != null ||
```

Comment:
We should probably check this also in the analytics collection phase

Reply:
This is about SGs. We've now moved this to the weaver.